commands/usernote.js

```diff
@@ -62,7 +70,20 @@ module.exports = {
     };
     let args = parseArgs(preparsedArgs, parseSettings);
     let command = args._.length ? args._[0].toLowerCase() : 'show';
-    args.subreddit = args.link || args.subreddit ? args.subreddit : 'pokemontrades';
+    if (!args.subreddit) {
+      if (moduleConfig.defaultSubreddit) args.subreddit = moduleConfig.defaultSubreddit;
+      if (args.link) {
+        const linkSubMatch = args.link.match(/reddit\.com\/r\/([^\/]+)/);
+        if (linkSubMatch) {
+          args.subreddit = linkSubMatch[1];
+        } else if (args.link.match(/reddit\.com\/message\//)) {
+          args.subreddit = undefined;
+        }
+      }
+    }
+    if (moduleConfig && moduleConfig.channelPermissions && (!moduleConfig.channelPermissions[channel] || (!_.isBoolean(moduleConfig.channelPermissions[channel]) && (args.subreddit===undefined || moduleConfig.channelPermissions[channel].findIndex(sub => sub.toLowerCase() === args.subreddit.toLowerCase()) < 0)))) {
+      throw {error_message: 'Access denied.'};
```

Comment:
I'm not sure about this check. It appears to be valid (in that I can't think of any input off the top of my head that would be able to get around it), but right now it's tied to our argument parsing logic. If we make changes to that in the future (e.g. accepting different types of reddit.com URLs), then we might unexpectedly end up with security vulnerabilities. This has no obvious security bugs, but I would prefer it if it obviously had no security bugs.

I think it would be better to have a lower-level check for this instead. For example, a check on `args.subreddit` [here](https://github.com/pokemontrades/porygon-bot/blob/3bf275fee0979843819f928493c08c7ff84ff114/commands/usernote.js#L73) would ensure that the subreddit is valid *after* all the argument parsing/fetching/etc. is finished. (This validation would probably have to be repeated for all calls to `usernoteHelper`, so maybe it would be good to have a helper function to do the validation.)